state-synchronizer/src/coordinator.rs

```diff
@@ -202,15 +209,21 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
         upstream_config: UpstreamConfig,
         executor_proxy: T,
         initial_state: SynchronizationState,
-    ) -> Self {
+    ) -> Result<Self> {
```

Comment:
as this is a constructor, what about panicking instead of returning an error on overflow?

Reply:
I thought about it.. don't feel super strongly. Right now, the only production code that calls this constructor panics if something goes wrong, so it's somewhat equivalent 😄 Personally, I prefer to leave it as-is so that future code will be forced to think about the failure if it calls this.